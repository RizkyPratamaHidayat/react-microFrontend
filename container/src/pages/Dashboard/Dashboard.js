import React, { lazy } from "react";
import clsx from "clsx";
import { makeStyles } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import Drawer from "@material-ui/core/Drawer";
import Box from "@material-ui/core/Box";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import List from "@material-ui/core/List";
import Typography from "@material-ui/core/Typography";
import Divider from "@material-ui/core/Divider";
import IconButton from "@material-ui/core/IconButton";
import Badge from "@material-ui/core/Badge";
import Container from "@material-ui/core/Container";
import { connect } from "react-redux";
import Grid from "@material-ui/core/Grid";
import Paper from "@material-ui/core/Paper";
import MenuIcon from "@material-ui/icons/Menu";
import ChevronLeftIcon from "@material-ui/icons/ChevronLeft";
import NotificationsIcon from "@material-ui/icons/Notifications";
import { Switch, Route, Link, BrowserRouter } from "react-router-dom";
import { ListItem, ListItemIcon, ListItemText } from "@material-ui/core";
import AssignmentTurnedInIcon from "@material-ui/icons/AssignmentTurnedIn";
import GroupIcon from "@material-ui/icons/Group";
import BookIcon from "@material-ui/icons/Book";
import LaptopIcon from "@material-ui/icons/Laptop";
import TimelineIcon from "@material-ui/icons/Timeline";
const StaffSite = lazy(() => import("../StaffSiteApp"));

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
  },
  toolbar: {
    paddingRight: 24, // keep right padding when drawer closed
  },
  toolbarIcon: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    padding: "0 8px",
    ...theme.mixins.toolbar,
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(["width", "margin"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
  },
  appBarShift: {
    marginLeft: drawerWidth,
    width: `calc(100% - ${drawerWidth}px)`,
    transition: theme.transitions.create(["width", "margin"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  menuButton: {
    marginRight: 36,
  },
  menuButtonHidden: {
    display: "none",
  },
  title: {
    flexGrow: 1,
  },
  drawerPaper: {
    position: "relative",
    whiteSpace: "nowrap",
    width: drawerWidth,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerPaperClose: {
    overflowX: "hidden",
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    width: theme.spacing(7),
    [theme.breakpoints.up("sm")]: {
      width: theme.spacing(9),
    },
  },
  appBarSpacer: theme.mixins.toolbar,
  content: {
    flexGrow: 1,
    height: "100vh",
    overflow: "auto",
  },
  container: {
    paddingTop: theme.spacing(4),
    paddingBottom: theme.spacing(4),
  },
  paper: {
    padding: theme.spacing(2),
    display: "flex",
    overflow: "auto",
    flexDirection: "column",
  },
  fixedHeight: {
    height: 240,
  },
}));

function Dashboard(props) {
  const { user } = props;
  const classes = useStyles();
  const [open, setOpen] = React.useState(true);
  const handleDrawerOpen = () => {
    setOpen(true);
  };
  const handleDrawerClose = () => {
    setOpen(false);
  };
  const callbackReduxState = {
    user
  }
  const drawer = (
    <div>
      <div className={classes.toolbar} />
      <Divider />
      <List>
        {user?.menu?.map((e, idx) => {
          var icons;
          if(e.name === 'bootcamp'){
            icons =  <LaptopIcon />
          }
          else if(e.name === 'syllabus'){
            icons =  <BookIcon />
          }
          else if(e.name === 'task'){
            icons =  <AssignmentTurnedInIcon />
          }   else if(e.name === 'stage'){
            icons =  <TimelineIcon />
          }
          else{
            icons =  <GroupIcon />
          }
          return (
            <li key={idx}>
              <ListItem button component={Link} to={e.name.replace(' ', '-')}>
                <ListItemIcon>
                  {icons}
                </ListItemIcon>
                <ListItemText primary={e.name} />
              </ListItem>
            </li>
          );
        })}
      </List>
    </div>
  );

  return (
    <div className={classes.root}>
      <CssBaseline />
      <AppBar
        position="absolute"
        className={clsx(classes.appBar, open && classes.appBarShift)}
      >
        <Toolbar className={classes.toolbar}>
          <IconButton
            edge="start"
            color="inherit"
            aria-label="open drawer"
            onClick={handleDrawerOpen}
            className={clsx(
              classes.menuButton,
              open && classes.menuButtonHidden
            )}
          >
            <MenuIcon />
          </IconButton>
          <Typography
            component="h1"
            variant="h6"
            color="inherit"
            noWrap
            className={classes.title}
          >
            Dashboard
          </Typography>
          <IconButton color="inherit">
            <Badge badgeContent={4} color="secondary">
              <NotificationsIcon />
            </Badge>
          </IconButton>
        </Toolbar>
      </AppBar>
      <BrowserRouter>
        <Drawer
          variant="permanent"
          classes={{
            paper: clsx(classes.drawerPaper, !open && classes.drawerPaperClose),
          }}
          open={open}
        >
          <div className={classes.toolbarIcon}>
            <IconButton onClick={handleDrawerClose}>
              <ChevronLeftIcon />
            </IconButton>
          </div>
          <List>{drawer}</List>
        </Drawer>
        <main className={classes.content}>
          <div className={classes.appBarSpacer} />
          <Container maxWidth="lg" className={classes.container}>
            <Switch>
              <Route
                exact
                path="/member-management"
                render={() => <StaffSite callbackReduxState={callbackReduxState} />}
              />
              <Route
                path="/bootcamp"
                render={() => <div>Bootcamp content</div>}
              />
              <Route
                path="/syllabus"
                render={() => <div>Syllabus content</div>}
              />
              <Route path="/task" render={() => <div>Task content</div>} />
              <Route path="/stage" render={() => <div>Stage content</div>} />
            </Switch>
          </Container>
        </main>
      </BrowserRouter>
    </div>
  );
}
const mapStateToProps = (state) => {
  return {
    user: state.authorizationReducer.user,
  };
};
export default connect(mapStateToProps)(Dashboard);
