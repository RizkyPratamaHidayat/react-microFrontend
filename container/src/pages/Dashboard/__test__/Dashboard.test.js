import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Dashboard from '../../Dashboard';

const renderer  = new ShallowRenderer();
describe('Test Dashboard Page', () => {
  it('render page properly', () => {
    renderer.render(<Dashboard />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});