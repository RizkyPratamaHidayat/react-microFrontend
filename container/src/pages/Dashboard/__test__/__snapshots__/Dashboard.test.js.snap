// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dashboard Page render page properly 1`] = `
<div
  className="makeStyles-root-1"
>
  <ForwardRef(WithStyles(CssBaseline)) />
  <ForwardRef(WithStyles(ForwardRef(AppBar)))
    className="makeStyles-appBar-4 makeStyles-appBarShift-5"
    position="absolute"
  >
    <ForwardRef(WithStyles(ForwardRef(Toolbar)))
      className="makeStyles-toolbar-2"
    >
      <ForwardRef(WithStyles(ForwardRef(IconButton)))
        aria-label="open drawer"
        className="makeStyles-menuButton-6 makeStyles-menuButtonHidden-7"
        color="inherit"
        edge="start"
        onClick={[Function]}
      >
        <UNDEFINED />
      </ForwardRef(WithStyles(ForwardRef(IconButton)))>
      <ForwardRef(WithStyles(ForwardRef(Typography)))
        className="makeStyles-title-8"
        color="inherit"
        component="h1"
        noWrap={true}
        variant="h6"
      >
        Dashboard
      </ForwardRef(WithStyles(ForwardRef(Typography)))>
      <ForwardRef(WithStyles(ForwardRef(IconButton)))
        color="inherit"
      >
        <ForwardRef(WithStyles(ForwardRef(Badge)))
          badgeContent={4}
          color="secondary"
        >
          <UNDEFINED />
        </ForwardRef(WithStyles(ForwardRef(Badge)))>
      </ForwardRef(WithStyles(ForwardRef(IconButton)))>
    </ForwardRef(WithStyles(ForwardRef(Toolbar)))>
  </ForwardRef(WithStyles(ForwardRef(AppBar)))>
  <BrowserRouter>
    <ForwardRef(WithStyles(ForwardRef(Drawer)))
      classes={
        Object {
          "paper": "makeStyles-drawerPaper-9",
        }
      }
      open={true}
      variant="permanent"
    >
      <div
        className="makeStyles-toolbarIcon-3"
      >
        <ForwardRef(WithStyles(ForwardRef(IconButton)))
          onClick={[Function]}
        >
          <UNDEFINED />
        </ForwardRef(WithStyles(ForwardRef(IconButton)))>
      </div>
      <ForwardRef(WithStyles(ForwardRef(List)))>
        <div>
          <div
            className="makeStyles-toolbar-2"
          />
          <ForwardRef(WithStyles(ForwardRef(Divider))) />
          <ForwardRef(WithStyles(ForwardRef(List)))>
            <ForwardRef(WithStyles(ForwardRef(ListItem)))
              button={true}
              component={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              to="/"
            >
              <ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
                <UNDEFINED />
              </ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
              <ForwardRef(WithStyles(ForwardRef(ListItemText)))
                primary="Member Management"
              />
            </ForwardRef(WithStyles(ForwardRef(ListItem)))>
            <ForwardRef(WithStyles(ForwardRef(ListItem)))
              button={true}
              component={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              to="/bootcamp"
            >
              <ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
                <UNDEFINED />
              </ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
              <ForwardRef(WithStyles(ForwardRef(ListItemText)))
                primary="Bootcamp"
              />
            </ForwardRef(WithStyles(ForwardRef(ListItem)))>
            <ForwardRef(WithStyles(ForwardRef(ListItem)))
              button={true}
              component={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              to="/syllabus"
            >
              <ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
                <UNDEFINED />
              </ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
              <ForwardRef(WithStyles(ForwardRef(ListItemText)))
                primary="Syllabus"
              />
            </ForwardRef(WithStyles(ForwardRef(ListItem)))>
            <ForwardRef(WithStyles(ForwardRef(ListItem)))
              button={true}
              component={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              to="/task"
            >
              <ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
                <UNDEFINED />
              </ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
              <ForwardRef(WithStyles(ForwardRef(ListItemText)))
                primary="Task"
              />
            </ForwardRef(WithStyles(ForwardRef(ListItem)))>
            <ForwardRef(WithStyles(ForwardRef(ListItem)))
              button={true}
              component={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "propTypes": Object {
                    "innerRef": [Function],
                    "onClick": [Function],
                    "replace": [Function],
                    "target": [Function],
                    "to": [Function],
                  },
                  "render": [Function],
                }
              }
              to="/stage"
            >
              <ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
                <UNDEFINED />
              </ForwardRef(WithStyles(ForwardRef(ListItemIcon)))>
              <ForwardRef(WithStyles(ForwardRef(ListItemText)))
                primary="Stage"
              />
            </ForwardRef(WithStyles(ForwardRef(ListItem)))>
          </ForwardRef(WithStyles(ForwardRef(List)))>
        </div>
      </ForwardRef(WithStyles(ForwardRef(List)))>
    </ForwardRef(WithStyles(ForwardRef(Drawer)))>
    <main
      className="makeStyles-content-12"
    >
      <div
        className="makeStyles-appBarSpacer-11"
      />
      <ForwardRef(WithStyles(ForwardRef(Container)))
        className="makeStyles-container-13"
        maxWidth="lg"
      >
        <Switch>
          <Route
            exact={true}
            path="/"
            render={[Function]}
          />
          <Route
            path="/bootcamp"
            render={[Function]}
          />
          <Route
            path="/syllabus"
            render={[Function]}
          />
          <Route
            path="/task"
            render={[Function]}
          />
          <Route
            path="/stage"
            render={[Function]}
          />
        </Switch>
      </ForwardRef(WithStyles(ForwardRef(Container)))>
    </main>
  </BrowserRouter>
</div>
`;
